Rubberduck.Core/UI/CodeExplorer/Commands/DeleteCommand.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Rubberduck.Interaction;
+using Rubberduck.Navigation.CodeExplorer;
+using Rubberduck.VBEditor.ComManagement;
+using Rubberduck.VBEditor.SafeComWrappers;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.UI.CodeExplorer.Commands
+{
+    public class DeleteCommand : CodeExplorerCommandBase
+    {
+        private readonly RemoveCommand _removeCommand;
+        private readonly IProjectsProvider _projectsProvider;
+        private readonly IMessageBox _messageBox;
+        private readonly IVBE _vbe;
+
+        public DeleteCommand(RemoveCommand removeCommand, IProjectsProvider projectsProvider, IMessageBox messageBox, IVBE vbe)
+        {
+            _removeCommand = removeCommand;
+            _projectsProvider = projectsProvider;
+            _messageBox = messageBox;
+            _vbe = vbe;
+
+            AddToCanExecuteEvaluation(SpecialEvaluateCanExecute);
+        }        
+
+        private bool SpecialEvaluateCanExecute(object parameter)
+        {
+            return _vbe.Kind == VBEKind.Standalone &&
+                   _removeCommand.CanExecute(parameter);
+        }
+
+        protected override void OnExecute(object parameter)
+        {
+            if (!(parameter is CodeExplorerComponentViewModel node) || node.Declaration is null)
+            {
+                return;
+            }
+
+            var qualifiedModuleName = node.Declaration.QualifiedName.QualifiedModuleName;
+            var component = _projectsProvider.Component(qualifiedModuleName);
+            if (component is null)
+            {
+                return;
+            }
+
+            // Permanent operation, prompt for confirm
+            var message = string.Format(Resources.CodeExplorer.CodeExplorerUI.ConfirmBeforeDelete_Prompt, qualifiedModuleName.Name);
+            if (!_messageBox.ConfirmYesNo(message, Resources.CodeExplorer.CodeExplorerUI.ConfirmBeforeDelete_Caption))
+            {
+                return;
+            }
+
+            // Have to build the file list *before* removing the component!
+            var files = new List<string>();
+            for (short i = 1; i <= component.FileCount; i++)
+            {
+                var fileName = component.GetFileName(i);
+                if (fileName != null) // Unsaved components have a null filename for some reason
+                {
+                    files.Add(component.GetFileName(i));
+                }
+            }
+
+            if (!_removeCommand.RemoveComponent(qualifiedModuleName, false))
+            {
+                return; // Remove was cancelled or failed
```

Comment:
No User-notification here?

Reply:
Already handled in the method. Updated comment.